app/db/AtomWorkshopDB.scala

```diff
@@ -37,24 +35,51 @@ object AtomWorkshopDB {
     }
   }
 
-  def getAtom(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom], atomType: AtomType, id: String) = {
+  def getAtom(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom], atomType: AtomType, id: String): Either[AtomAPIError, Atom] = {
     transformAtomLibResult(datastore.getAtom(AtomWorkshopDB.buildKey(atomType, id)))
   }
 
-  def updateAtom(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom], atomType: AtomType, user: User, currentVersion: Atom, newAtom: Atom): Either[AtomAPIError, Unit]  = {
-    val updatedAtom = currentVersion.copy(
-      contentChangeDetails = buildContentChangeDetails(user, Some(currentVersion.contentChangeDetails), updateLastModified = true),
-      defaultHtml = buildDefaultHtml(atomType, currentVersion.data, Some(currentVersion.defaultHtml)),
-      data = newAtom.data
-    )
+  private def updateAtomInDatastore(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom], atom: Atom): Either[AtomAPIError, Unit] = {
     try {
-      val result = datastore.updateAtom(updatedAtom)
-      Logger.info(s"Successfully updated atom of type ${atomType.name} with id ${currentVersion.id}")
-      Right(transformAtomLibResult(result))
+      val result = datastore.updateAtom(atom)
+      Logger.info(s"Successfully updated atom of type ${atom.atomType.name} with id ${atom.id}")
+      transformAtomLibResult(result)
     } catch {
       case e: Exception => processException(e)
     }
+  }
+
+  private def createAtomFromUpdatedAtom(atom: Atom, updatedAtom: Atom, user: User): Atom =
+    atom.copy(
+      contentChangeDetails = buildContentChangeDetails(user, Some(atom.contentChangeDetails), updateLastModified = true),
+      defaultHtml = buildDefaultHtml(atom.atomType, updatedAtom.data, Some(atom.defaultHtml)),
+      data = updatedAtom.data
+    )
+
+  def updateAtom(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom],
+                 atomType: AtomType,
+                 user: User,
+                 currentVersion: Atom,
+                 newAtom: Atom): Either[AtomAPIError, Unit] = {
+
+    val updatedAtom: Atom = createAtomFromUpdatedAtom(currentVersion, newAtom, user)
+    updateAtomInDatastore(datastore, updatedAtom)
+  }
+
+  def updateAtomByPath(datastore: DynamoDataStore[_ >: ExplainerAtom with CTAAtom with MediaAtom],
+                       atomType: AtomType,
+                       user: User,
+                       currentJson: Json,
+                       newJson: Json): Either[AtomAPIError, Unit]  = {
+
+    val updatedAtomJson: Json = currentJson.deepMerge(newJson)
+
+    val updatedAtom = for {
+      atom <- parseStringToAtom(currentJson.toString)
```

Comment:
You don't need to stringify the `Json`.
Instead you can do: `currentJson.as[Atom]`